Exercise 2 Multiplexer/multiplexer-rt/src/com/tridiumuniversity/multiplexer/BMultiplexer.java

```diff
@@ -0,0 +1,256 @@
+  
+  /**
+   * Get the {@code in3} property.
+   * @see #in3
+   */
+  public boolean getIn3() { return getBoolean(in3); }
+  
+  /**
+   * Set the {@code in3} property.
+   * @see #in3
+   */
+  public void setIn3(boolean v) { setBoolean(in3, v, null); }
+
+////////////////////////////////////////////////////////////////
+// Property "s0"
+////////////////////////////////////////////////////////////////
+  
+  /**
+   * Slot for the {@code s0} property.
+   * @see #getS0
+   * @see #setS0
+   */
+  public static final Property s0 = newProperty(Flags.SUMMARY, ((BBoolean)(BBoolean.DEFAULT)).getBoolean(), null);
+  
+  /**
+   * Get the {@code s0} property.
+   * @see #s0
+   */
+  public boolean getS0() { return getBoolean(s0); }
+  
+  /**
+   * Set the {@code s0} property.
+   * @see #s0
+   */
+  public void setS0(boolean v) { setBoolean(s0, v, null); }
+
+////////////////////////////////////////////////////////////////
+// Property "s1"
+////////////////////////////////////////////////////////////////
+  
+  /**
+   * Slot for the {@code s1} property.
+   * @see #getS1
+   * @see #setS1
+   */
+  public static final Property s1 = newProperty(Flags.SUMMARY, ((BBoolean)(BBoolean.DEFAULT)).getBoolean(), null);
+  
+  /**
+   * Get the {@code s1} property.
+   * @see #s1
+   */
+  public boolean getS1() { return getBoolean(s1); }
+  
+  /**
+   * Set the {@code s1} property.
+   * @see #s1
+   */
+  public void setS1(boolean v) { setBoolean(s1, v, null); }
+
+////////////////////////////////////////////////////////////////
+// Property "out"
+////////////////////////////////////////////////////////////////
+  
+  /**
+   * Slot for the {@code out} property.
+   * @see #getOut
+   * @see #setOut
+   */
+  public static final Property out = newProperty(Flags.SUMMARY, ((BBoolean)(BBoolean.DEFAULT)).getBoolean(), null);
+  
+  /**
+   * Get the {@code out} property.
+   * @see #out
+   */
+  public boolean getOut() { return getBoolean(out); }
+  
+  /**
+   * Set the {@code out} property.
+   * @see #out
+   */
+  public void setOut(boolean v) { setBoolean(out, v, null); }
+
+////////////////////////////////////////////////////////////////
+// Type
+////////////////////////////////////////////////////////////////
+  
+  @Override
+  public Type getType() { return TYPE; }
+  public static final Type TYPE = Sys.loadType(BMultiplexer.class);
+
+/*+ ------------ END BAJA AUTO GENERATED CODE -------------- +*/
+
+  @Override
+  public void changed(Property property, Context context)
+  {
+    if(s0.equals(property)|| s1.equals(property) || in0.equals(property) || in1.equals(property) || in2.equals(property)|| in3.equals(property))
+    {
+      update();
+    }
+    super.changed(property, context);
+  }
+
+  public void update()
+  {
+    if(getS0() && getS1()){
+      setOut(getIn3());
+    }
+    else if (!getS0()&&getS1())
+    {
+      setOut(getIn1());
```

Comment:
I think you got two rows of the truth table swapped. I think this should be `setOut(getIn2())` and line 248 should be `setOut(getIn1())`.